runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1062,22 +1073,52 @@ TR_J9ServerVM::allocateCodeMemory(TR::Compilation * comp, uint32_t warmCodeSize,
 bool
 TR_J9ServerVM::sameClassLoaders(TR_OpaqueClassBlock * class1, TR_OpaqueClassBlock * class2)
    {
+   void *class1Loader = nullptr;
+   void *class2Loader = nullptr;
+   do
       {
-      OMR::CriticalSection getRemoteROMClass(_compInfoPT->getClientData()->getROMMapMonitor());
-      auto it_class1 = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) class1);
-      if (it_class1 != _compInfoPT->getClientData()->getROMClassMap().end())
          {
-         auto it_class2 = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) class2);
-         if (it_class2 != _compInfoPT->getClientData()->getROMClassMap().end())
+         OMR::CriticalSection getRemoteROMClass(_compInfoPT->getClientData()->getROMMapMonitor());
```

Comment:
I suggest using a helper that finds the classloader of a j9method. This could be called twice here and once in `isClassLoadedBySystemClassLoader`